### ISSUE TITLE
RuboCop auto-corrected code contains too many deep indentation

### ISSUE THREAD

**Issue by tompng** [rubocop/rubocop#13039]
## Is your feature request related to a problem? Please describe.

RuboCop auto-corrected code contains deep indentation like this:
```ruby
aa = { long_key1: long_value1,
       long_key2: long_value2 }
foo.bar(long_arg1,
        long_arg2)
foobar = case a
         when b
           c
         when d
           e
         end
```
These indentation is allowed, but I think it is better to avoid generating deep indentation by autocorrect.
Otherwise, some rubyists might wrongly think that deeper indentation is the preferred style.
I think these deep indentation styles are too frequently used in real world under the influence of autocorrect.

Hashes
```ruby
a = { too_long1: value1, too_long2: value2 }

# auto-corrected style
# 124 patterns in rails/rails, found by regexp /^( +)[^ ].+\{.+,\n\1 {3}/
a = { too_long1: value1,
      too_long2: value2 }

# hand-written style
# 1430 patterns in rails/rails, found by regexp /^( +)[^ ].+\{\n\1 {2}[^ ]/
a = {
  too_long1: value1,
  too_long2: value2
}
```

Method arguments
```ruby
some.method(too_long_arg1, too_long_arg2)

# auto-corrected style
# 233 patterns in rails/rails, found by regexp /^( +)[^ ].+\(.+,\n\1 {3,}/
some.method(too_long_arg1,
            too_long_arg2)

# hand-written style
# 1383 pattersn in rails/rails, found by regexp /^( +)[^ ].+\(\n\1 {2}[^ ]/
some.method(
  too_long_arg1,
  too_long_arg2
)
```

Conditional assignments
```ruby
if cond
  ab = c
else
  ab = d
end

# auto-corrected style
# 7 patterns in rails/rails, found by regexp /^( +)[^ ].+= (if|case|begin).*\n\1 {3,}/
ab = if cond
       c
     else
       d
     end

# Also a good code example written in Layout/AssignmentIndentation
# 53 patterns in rails/rails, found by regexp /^( +)[^ ].+=\n\1 {2}(if|case|begin)/
ab =
  if cond
    c
  else
    d
  end
```

## Describe the solution you'd like

Autocorrect with shallower indent by default

## Describe alternatives you've considered

Add a missing shallower good example to cop documentation, for example, https://docs.rubocop.org/rubocop/cops_style.html#enforcedstyle-assign_to_condition-default
```ruby
# good
bar = if foo
        1
      else
        2
      end

# good (this example is missing)
bar =
  if foo
    1
  else
    2
  end
```

**Comment by dvandersluis** [comment]
Autocorrect for `Style` (and some `Layout`) cops follows the `EnforcedStyle` setup in configuration. If there's a specific bug feel free to open but for now, RuboCop is just doing what it's configured to do.
